sdk/eventgrid/Azure.Messaging.EventGrid/EventGridSourceGenerator/EventGridSourceGenerator.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Text;
+
+namespace EventGridSourceGenerator
+{
+    /// <summary>
+    /// This class generates the SystemEventNames constant values as well as the SystemEventExtensions which contains a mapping
+    /// from constant values to deserialization method for each system event.
+    /// </summary>
+    [Generator]
+    internal class EventGridSourceGenerator : ISourceGenerator
+    {
+        private SourceVisitor _visitor;
+        private const string Indent = "    ";
+
+        public void Execute(GeneratorExecutionContext context)
+        {
+            // using the context, get a list of syntax trees in the compilation
+            var syntaxTrees = context.Compilation.WithAssemblyName("Azure.Messaging.EventGrid").SyntaxTrees;
+            _visitor = new SourceVisitor();
+
+            foreach (CSharpSyntaxTree tree in syntaxTrees)
+            {
+                _visitor.Visit(tree.GetCompilationUnitRoot());
+            }
+
+            context.AddSource("SystemEventNames", SourceText.From(ConstructSystemEventNames(), Encoding.UTF8));
```

Comment:
```suggestion
            context.AddSource("SystemEventNames.cs", SourceText.From(ConstructSystemEventNames(), Encoding.UTF8));
```

Reply:
Hmm I'm not sure this is needed as it is a "hint" rather than the actual path.